spec/features/settings/dex_connector_oidc_feature_spec.rb

```diff
@@ -0,0 +1,142 @@
+require "rails_helper"
+
+# rubocop:disable RSpec/ExampleLength
+# TODO: do we need js?
+# describe "Feature: OIDC connector settings", js: true do
+describe "Feature: OIDC connector settings", js: true do
+  let!(:user) { create(:user) }
+  let!(:dex_connector_oidc) do
+    VCR.use_cassette("oidc/validate_connector", allow_playback_repeats: true, record: :none) do
```

Comment:
You don't need to use VCR when calling factories to create objects. VCR should be used to wrap a line or a block of code that interacts with external HTTP requests.

Reply:
Just realized that you've created custom validators for attributes that indeed make external HTTP requests and that's probably the reason you added this. A workaround for this would be to skip validation when creating via factories.

I wonder if it's a good idea to have a custom validator for attributes making external HTTP requests. I'd appreciate if someone else with more experience could give an input about it.

Reply:
That is indeed why it's there. :)

I guessed that the cost of littering VCR wrappers all around the test suite was a worthwhile trade for putting the data validation as close as possible to the database.  Velum's future isn't completely certain, so part of my reasoning was that doing all validation in one place would provide for cleaner support of an alternate interface which interacted via Velum programmatically through the web API rather than through the HTML, for example.  The other reasoning is that this is more reliable and less brittle than a client-side solution.

To explain why the external http call is made to begin with:  the validation duplicates the check Dex makes when it's started (requests webfinger data and compares the issuer).  If this value is inaccurate in the database and gets pushed out to the Dex config, orchestration will fail because Dex will fail to restart.  We therefore want to do as much as possible to ensure that the database never receives data which could cause that orchestration failure.  Longer-term, it might make sense to additionally do a check when updating the Dex config, but that's _way_ beyond the scope of this PR. :)

In any event, since the external call must be made, it makes sense to me to do that in the testing as well. Removing it seems like it would decrease the validity of the testing.

Reply:
If someone can provide a better way to do this particular test without decreasing the validity of the tests, I'm definitely all for it.  My Ruby is not particularly strong, so there was not much finesse used in this test code. :)

Reply:
The workaround suggestion, which might not be the final solution, does not decrease the validity of the tests. Actually, it will decrease the verbosity and enable us to write better tests and focus on what is important to be tested. 

The purpose of `create(:dex_connector_oidc)` is to seed the database with a record. We can assume and ensure that the data produced by the factory is valid. This way bypassing the validation wouldn't be an issue. Here's an example [0] of what I was suggesting. That would fix the issue with `create(:dex_connector_oidc)` but wouldn't fix the issue of `FactoryGirl.lint` inside `DatabaseCleaner.cleaning`.

[0] https://stackoverflow.com/a/45965846

Reply:
Good point; the validation is explicitly tested elsewhere, and these instances are all really used to test something other than creation. 

Adding a trait to the factory (thanks for the example) allowed a number of creations to be cleaned up.  In the process, I noticed that the pillar test cases were written in such a way that they'd always pass, so that's also been cleaned up now.

I'm not sure that it's a problem to have the `lint` done inside datbaase cleaner actually run validation.  That's why the lint is there, right - to validate that the factories work?  One could exclude this particular factory from the normal linting, and instead run it explicitly with the validation skipped.  But IMHO that's quite a bit uglier than just leaving the VCR cassette definition in the database cleaner. :)